private/bufpkg/bufanalysis/bufanalysis.go

```diff
@@ -200,34 +196,28 @@ func PrintFileAnnotations(writer io.Writer, fileAnnotations []FileAnnotation, fo
 	if err != nil {
 		return err
 	}
-	for _, fileAnnotation := range fileAnnotations {
-		s, err := FormatFileAnnotation(fileAnnotation, format)
-		if err != nil {
-			return err
-		}
-		if _, err := writer.Write([]byte(s + "\n")); err != nil {
-			return err
-		}
-	}
-	return nil
+	return formatToPrinter[format].PrintFileAnnotations(
+		writer,
+		fileAnnotations,
+	)
 }
 
 // FormatFileAnnotation formats the FileAnnotation.
 func FormatFileAnnotation(fileAnnotation FileAnnotation, format Format) (string, error) {
+	buffer := bytes.NewBuffer(nil)
 	switch format {
 	case FormatText:
-		return fileAnnotation.String(), nil
+		_ = printFileAnnotationAsText(buffer, fileAnnotation)
```

Comment:
You can't ignore this error, you need to propagate it

Reply:
Even if `printFileAnnotationAsText` always returns `nil` similar to `bytes.Buffer`?

Reply:
If it will always return nil, then the function should not have an error signature. I think what you mean is that writes to the `buffer` will never result in error, so this function, due to how we happen to be calling it, will always return nil. That's a big usage assumption IMO - if we later change from `bytes.Buffer` to something else that can error, we are then ignoring an error. We should always check errors such as these.